model/task_lifecycle.go

```diff
@@ -303,7 +278,28 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
 		return errors.Wrap(checkResetSingleHostTaskGroup(t, caller), "resetting single host task group")
 	}
 
-	return errors.WithStack(resetTask(t.Id, caller, false))
+	return errors.WithStack(resetTask(t, caller, false))
+}
+
+// resetTask attempts to archive the task, resets it, and resets the TaskCache in the build as well.
+func resetTask(t *task.Task, caller string, logIDs bool) error {
```

Comment:
I think the reason this function gets the task again is to make sure it's getting the full task document, whereas the task we might be passing in could have only partial fields, meaning we'd do a partial archive (for example, with dependencies in SetActiveState)